src/utils/typedData.ts

```diff
@@ -64,11 +64,41 @@ const revisionConfiguration: Record<Revision, Configuration> = {
   },
 };
 
+/**
+ * Asserts that the given data is within the specified range.
+ *
+ * @param {unknown} data - The data to check.
+ * @param {string} type - The type of the data.
+ * @param {Object} range - The range object containing min and max values.
+ * @param {bigint} range.min - The minimum value of the range.
+ * @param {bigint} range.max - The maximum value of the range.
+ * @throws Will throw an error if the data is out of the specified range.
+ * 
+ * @example
+ * ```typescript
+ * assertRange('10', 'felt', { min: BigInt(0), max: BigInt(100) }); // No error
+ * assertRange('200', 'felt', { min: BigInt(0), max: BigInt(100) }); // Error: 200 (felt) is out of bounds [0, 100]
```

Comment:
c/p from another issue:
As explained in Telegram https://t.me/c/2066195975/59 ,
please follow very rigorously the template of this PR https://github.com/starknet-io/starknet.js/pull/1100, especially for the examples : result always in a result variable, result displayed in a comment (//);